### PR TITLE
[MM-12155] Create new SendCtrlEnterSection component from AdvancedSettingsDisplay

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Create new SendCtrlEnterSection component from AdvancedSettingsDisplay

Also, use constant for 'code_block_ctrl_enter' from mattermost-redux

#### Ticket Link
Jira ticket: [MM-12155](https://mattermost.atlassian.net/browse/MM-12155)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -489,7 +489,6 @@
       2,
       "never"
     ],
-    "react/jsx-filename-extension": 2,
```
Comment (saturninoabril) [inline .eslintrc.json]: Changed this to allow .js file extension of React component

File: components/user_settings/advanced/send_ctrl_enter_section/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import {Preferences} from 'mattermost-redux/constants';
+import {get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
+import SendCtrlEnterSection from './send_ctrl_enter_section';
+
+function mapStateToProps(state) {
+    return {
+        currentUserId: getCurrentUserId(state),
+        sendMessageOnCtrlEnter: getPreference(state, Preferences.CATEGORY_ADVANCED_SETTINGS, Preferences.ADVANCED_SEND_ON_CTRL_ENTER, 'false'),
```
Comment (enahum) [inline components/user_settings/advanced/send_ctrl_enter_section/index.js]: why this property is a string and not a boolean?

**Comment by enahum** [comment]
@saturninoabril what else needs to be done here? maybe if you add more stuff to this PR ask for my review again

**Comment by saturninoabril** [comment]
I'm closing this PR temporarily and will reopen once I get back into this.

**Pull request closed without merging**